osu.Game.Tests/Visual/SongSelect/TestSceneBeatmapCarousel.cs

```diff
@@ -83,6 +83,41 @@ public void TestTraversal()
             waitForSelection(set_count, 3);
         }
 
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestTraversalHold(bool forwards)
+        {
+            var sets = new List<BeatmapSetInfo>();
+            const int create_this_many_sets = 200;
+
+            for (int i = 0; i < create_this_many_sets; i++)
+            {
+                var set = createTestBeatmapSet(i + 1);
+                sets.Add(set);
+            }
+
+            loadBeatmaps(sets);
+
+            void selectNextAndAssert(int amount)
+            {
+                setSelected(forwards ? 1 : create_this_many_sets, 1);
+                string text = forwards ? "Next" : "Previous";
+                AddStep($"{text} beatmap {amount} times", () =>
+                {
+                    for (int i = 0; i < amount; i++)
+                    {
+                        carousel.SelectNext(forwards ? 1 : -1);
+                    }
+                });
+                waitForSelection(forwards ? amount + 1 : create_this_many_sets - amount);
+            }
+
+            for (int i = 1; i < create_this_many_sets; i += i)
+            {
+                selectNextAndAssert(i);
+            }
```

Comment:
Not sure why this way was chosen to test this but I suppose it works...?

Reply:
#8470 (PR before this PR) had an optimization that would fail over different amounts of select next called. And old behaviour that relied on drawables also would fail over different amounts (and for some reason would always fail after you had simultaneously selected at least 7 times). Not sure if it will catch any regressions with the new logic tho.

Reply:
No, I get _what_ was tested, just not sure why the exponentially-increasing number of changes each step. It's just mostly my remark though, nothing worth changing really.

Reply:
The old optimization that didn't work (and thus wasn't commited) ramped up exponentially, that's why the test was exponential.